snmp/datadog_checks/snmp/models.py

```diff
@@ -56,11 +56,6 @@ def __init__(self, value):
 
     def resolve(self, mib_view_controller):
         # type: (MibViewController) -> None
-        if self._parts is not None:
-            # Client code should only call this if they're certain the
-            # underlying OID isn't resolved yet.
-            raise RuntimeError('Already resolved as {}'.format(self._parts))
```

Comment:
This is a bit tricky, but I think this check doesn't actually make sense… There are valid cases when `_parts` is already available but the `object_identity` has no MIB information yet.

Eg when `enforce_constraints=False`, `lookupMib=False` is passed to PySNMP, which results in it returning varbinds that have the raw OID as a string, eg `1.2.3.4...`. In that case `_parts` will already be set because `parse_as_oid_tuple()` will succeed. So this check would make `.resolve()` fail, even though the underlying `ObjectIdentity` is not MIB-resolved yet (which is what we actually want to do in this method).

So it seems we should just remove it - I had not taken this use case into account when originally introducing it in #6242.